### PR TITLE
fix the incorrect execution time of join in explain analyze

### LINKED ISSUE #5412: The execution time of join in explain analyze is not correct
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

origin_img_v2_698c973a-e685-4149-8fe8-55020c16f8ag

From https://github.com/pingcap/tiflash/blob/5d680c4287ed5c2c9c3feb9ca4938d3e351300a3/dbms/src/Flash/Coprocessor/DAGResponseWriter.cpp#L115-L136, we should add the execution time of `HashJoinBuildBlockInputStream` instead of streams of build_side_root_executor_id

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #5412

Problem Summary:

### What is changed and how it works?
- Add the execution time of HashJoinBuildBlockInputStream instead of streams of build_side_root_executor_id in `DAGReponseWriter::addExecuteSummaries`
- Add join_build_info in mpp_task_tracing

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Fix the incorrect execution time of join in explain analyze.
```

### CODE REVIEW

**Comment by SeaRise** [comment]
434690946278883329 dot

**Review by ywqzzy — APPROVED** [review]
LGTM

File: dbms/src/Flash/Coprocessor/DAGResponseWriter.cpp [hunk 1]
```diff
@@ -120,17 +120,13 @@ void DAGResponseWriter::addExecuteSummaries(tipb::SelectResponse & response, boo
...
-                        for (const auto & join_build_stream : build_side_it->second)
-                        {
-                            if (auto * p_stream = dynamic_cast<IProfilingBlockInputStream *>(join_build_stream.get()))
-                                process_time_for_build = std::max(process_time_for_build, p_stream->getProfileInfo().execution_time);
-                        }
-                        current.time_processed_ns += process_time_for_build;
+                        if (auto * p_stream = dynamic_cast<IProfilingBlockInputStream *>(join_build_stream.get()); p_stream)
+                            process_time_for_build = std::max(process_time_for_build, p_stream->getProfileInfo().execution_time);
                     }
+                    current.time_processed_ns += process_time_for_build;
```
Comment (windtalker) [inline dbms/src/Flash/Coprocessor/DAGResponseWriter.cpp:129]: Consider the case that the MPPTask has multiple joins, then it have to wait all build fininshed, does `join_build_streams` include all the join streams it had to wait?

Comment (SeaRise) [reply dbms/src/Flash/Coprocessor/DAGResponseWriter.cpp:129]: `dag_context.getExecutorIdToJoinIdMap().find(executor_id)` will save all join_executor_id in the child

**Pull request merged**